uiowa/uiowa
# Card background options, template updates

Resolves #2425
Resolves #2426 

Adds new bg color options to cards.

# How to test

## Testing the update hook

Sync site and/or import config `drush @site.env cim` or `blt ds --site=sandbox.uiowa.edu`

Existing cards on sites need to be tested that they receive the new defaults. After running the update hook:
- If the existing card had no background set (the standard white background), it should now have `White (default)` set as its background style
- If the existing card had a gray background, it should now have `Gray` set as its background style

## Testing color options for v3

```
rm -rf ./node_modules
yarn cache clean
yarn install
yarn workspace uids_base gulp --development
```
Sync site and/or import config `drush @site.env cim` or `blt ds --site=sandbox.uiowa.edu`

```
drush @sandbox.local cr
```
- Test that newly created cards begin with `White (default)` as their bg style option
- Test that upon saving, bg options including `White (default)` and others properly apply and appear upon editing the card
- Test event (v3), events (v3) and aggregator (v3) styles are all correct and use the card template (`card` classes) instead of the old event template (`events-card` classes).

## Testing sandbox pages for v3

- https://sandbox.prod.drupal.uiowa.edu/card-and-event-background-testing
- https://sandbox.prod.drupal.uiowa.edu/cards 
  - In each background color section, adjust the card background color to be `none` `white` `gray` `gold` `black`
  - Verify that color changes in layout builder preview mode and once page is saved. 
- https://sandbox.prod.drupal.uiowa.edu/articles
- https://sandbox.prod.drupal.uiowa.edu/people-directory
- https://sandbox.prod.drupal.uiowa.edu/events-listing
- https://sandbox.prod.drupal.uiowa.edu/event-combinations
- https://sandbox.prod.drupal.uiowa.edu/aggregator-0
- https://sandbox.prod.drupal.uiowa.edu/kitchen-sink
  - Review kitchen sink page color contrast
  - Add `gray` background to each text area section
  - Add `white` background option to each text area


## Testing color options for v2:
- `blt ds --site=hawkeyemarchingband.uiowa.edu`
- Add featured content, articles, events, people paragraph sections. 
  - Test with list, grid, masonry 
  - Test with pager and more button
  - Test with gray, gold, dark, and black background colors.
